### PR TITLE
Update message to show category

### PR DESCRIPTION
**Pull request by svr333**
## Description
This PR updates the current message so it shows which category something needs to be unlocked in

## Related Issues (if applicable)
discord suggestion #1477

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

**Review by TheBusyBiscuit — DISMISSED** [review]
Really no need for `String.format()` here, simple concatenation should be faster and easier

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/guide/SurvivalSlimefunGuide.java [hunk 1]
```diff
@@ -634,7 +634,7 @@ private void addBackButton(ChestMenu menu, int slot, Player p, PlayerProfile pro
                 return item;
             }
 
-            String lore = hasPermission(p, slimefunItem) ? "&fNeeds to be unlocked elsewhere" : "&fNo Permission";
+            String lore = hasPermission(p, slimefunItem) ? String.format("&fNeeds to be unlocked in %s", slimefunItem.getCategory().getDisplayName(p)) : "&fNo Permission";
```
Comment (svr333) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/guide/SurvivalSlimefunGuide.java:637]:
```suggestion
            String lore = hasPermission(p, slimefunItem) ? "&fNeeds to be unlocked in " + slimefunItem.getCategory().getDisplayName(p) : "&fNo Permission";
```

**Review by WalshyDev — APPROVED** [review]
LGTM

**Review by Sfiguz7 — APPROVED** [review]

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/guide/SurvivalSlimefunGuide.java [hunk 2]
```diff
@@ -634,7 +634,7 @@ private void addBackButton(ChestMenu menu, int slot, Player p, PlayerProfile pro
                 return item;
             }
 
-            String lore = hasPermission(p, slimefunItem) ? "&fNeeds to be unlocked elsewhere" : "&fNo Permission";
+            String lore = hasPermission(p, slimefunItem) ? "&fNeeds to be unlocked in " + slimefunItem.getCategory().getDisplayName(p) : "&fNo Permission";
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/guide/SurvivalSlimefunGuide.java:637]: Why wasn't this localized...?

**Pull request merged**